File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string s) {
    string answer = "";
    bool stchar = true; 
        
    for (char &c : s)
    {
        if (c == ' ')
        {
            stchar = true;
            continue;
        }
        if(stchar && c >= 'a' && c <= 'z')
            c -= 'a' - 'A';
        else if(!stchar && c >= 'A' && c <= 'Z')
            c += 'a' - 'A';
        stchar = false;
    }
    return s;
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.py
def solution(s):
    answer = ''
    for i in range(len(s)):
        if i == 0 and 'a' <= s[0] <= 'z':
            answer += s[0].upper()
        elif i-1 >= 0 and s[i-1] == ' ' and 'a' <= s[i] <= 'z':
            answer += s[i].upper()
        elif 'A' <= s[i] <= 'Z' and i-1 >= 0 and s[i-1] != ' ':
            answer += s[i].lower()
        else:
            answer += s[i]

    return answer